src/concrete_difference.jl

```diff
@@ -0,0 +1,69 @@
+export difference
+
+# alias for set difference
+import Base: \
+
+"""
+    \\(X::LazySet, Y::LazySet)
+
+Convenience alias for set difference.
+
+### Input
+
+- `X` -- a set
+- `Y` -- another set
+
+### Output
+
+The set difference between `X` and `Y`.
+"""
+\(X::LazySet, Y::LazySet) = difference(X, Y)
+
+# =================================
+# Set difference between intervals
+# =================================
+
+"""
+    difference(I1::Interval{N}, I2::Interval{N}) where {N}
+
+Return the set difference between the given intervals.
+
+The set difference is defined as:
+
+```math
+    I1 \\ I2 = \\{x: x ∈ I₁~\\textrm{and} x ∉ I₂ \\}
+```
+
+### Input
+
+- `I1` -- first interval
+- `I2` -- second interval
+
+### Output
+
+- An `EmptySet` if the set difference is empty.
+- A `UnionSetArray` of `Interval` sets otherwise.
```

Comment:
I would rather expext one of the following two options:
1. a `UnionSetArray` in all cases
2. an `EmptySet` or an `Interval` or a `Union` of two `Interval`s

Reply:
Ok, i thought about this. Maybe the second choice is nicer in the sense that the outputs are somehow "canonical". The drawback is that it can make the code that consumes the new `difference` function a bit trickier, namely the set difference between hyperrectangles. I was planning to write it by considering the set differences along each dimension then taking the cartesian product with the other dimensions (without considering the removed parts).

In sum i vote to keep option 2️⃣

Reply:
I do not think that your CP idea works in general.